### ISSUE TITLE
Possibility to make this into an app to check for updates / re-apply after update

### ISSUE THREAD

**Issue by ghost** [moosd/Needle#29]
Is it possible to put this into an application that will notify when a new update is available and that will notice when the patch is no longer applied due to an update, to then include a re-apply process?

**Comment by ale5000-git** [comment]
My fork of Needle is able to run also inside Android but patching automatically won't be so beneficial.
The patching process may take half an hour but also 4 hours on old phones.

Detecting if the framework.jar is patched or not I suppose is possbile but I think tricky.
It is in my ToDO list but it isn't a priority.

**Comment by ghost** [comment]
What is the purpose of your fork? To patch without needing a secondary system to hook up to via USB? As in, via APK?

**Comment by ale5000-git** [comment]
No, it is still Python code but it may run inside Android through QPython.
My fork can be executed everywhere in every variant of Python.
It allow to choose if patch through ADB (mode 1), patch local file (mode 2) or patch file from the phone (hidden mode 3).
Mode 1 tecnically can also patch from a phone another phone  through ADB (not tested).

It is [here](https://github.com/ale5000-git/tingle).
